lib/prompt-config.sh

```
@@ -0,0 +1,96 @@
+# Shared prompt settings loader for bash and zsh.
+# Expects DOTFILES_DIR to point at the repository root.
+
+# Resolve DOTFILES_DIR if the caller did not set it
+if [ -z "${DOTFILES_DIR:-}" ]; then
+    _prompt_cfg_src=""
+    if [ -n "${ZSH_VERSION:-}" ]; then
+        _prompt_cfg_src="${(%):-%x}"
```

Shellcheck (0.11.0)

[error] Parameter expansions can't start with (. Double check syntax.

(SC2296)